docs/addon_manager.md

```diff
@@ -0,0 +1,137 @@
+## Addons Management
```

Comment:
I'm thinking I start to put together a "tao of kops" or something - e.g. feels like kubectl, why versioned manifests, the apiserver, why are so many fields inferred, what is nodeup/protokube etc.  But a separate PR I think...